app/questionnaire/placeholder_parser.py

```diff
@@ -80,12 +79,19 @@ def __init__(
     def __call__(
         self, placeholder_list: Sequence[Mapping]
     ) -> MutableMapping[str, ValueSourceEscapedTypes | ValueSourceTypes]:
-        placeholder_list = QuestionnaireSchema.get_mutable_deepcopy(placeholder_list)
-
         sections_to_ignore = list(self._routing_path_block_ids_by_section_key)
+        source_type = "calculated_summary"
```

Comment:
I think setting source_type as calculated_summary for everything is a bit confusing, rather than using source_type, could potentially use the assess_routing_path bool? As if that is False we know to get the placeholder dependencies in the `get_routing_path_block_ids` method?

Reply:
Updated to use `assess_routing_path` instead of `source_type` here.